### ISSUE TITLE
Is it possible to use torch.linalg.cholesky for cholesky decomposition of a huge matrix that doesn't fit in memory?

### ISSUE THREAD

**Issue by mohamad-amin** [pytorch/pytorch#68696]
I guess the question speaks for itself.

I would like to actually do block cholesky decomposition such that each of the blocks of my huge matrix fit in the memory but the matrix itself doesn't fit.

Is it possible to do this using PyTorch?

**Comment by nikitaved** [comment]
No, it is not possible for generic matrices. If your matrix is block-diagonal, however, you can use `torch.linalg.cholesky` to factor the individual blocks.

**Comment by mohamad-amin** [comment]
Is there any plan on implementing this? There has been a couple of papers regarding how this is feasible (Used in GPyTorch, Falcon and etc) and how it scales. I basically will have to implement this most likely (personal-academic reasons), so would be glad to contribute if there is already a thread on implementing this.

**Comment by Lezcano** [comment]
There is no real plan to add these approximation algorithms to core PyTorch any time soon. It'll be best if you use the implementations in GPyTorch (or any other third party library) of conjugated gradient descent or the Lanzos method.
